src/power_grid_model/data_types.py

```diff
@@ -167,38 +167,40 @@ class SparseBatchColumnarData(TypedDict):
 Columnar data can be :class:`SingleColumnarData` or :class:`BatchColumnarData`.
 """
 
+_SingleComponentData = TypeVar("_SingleComponentData", SingleArray, SingleColumnarData)  # deduction helper
```

Comment:
i was only able to find this precisely because it suddenly started complaining on mypy on the `scripts/` dir when running locally. this almost broke mypy for our users